### DISCUSSION TITLE
IMG.CFG and custom formats.

### DISCUSSION THREAD

**Discussion by prime6809** [keirf/flashfloppy#654] [Q&A]
Posted a version of this to the FB group, so you may have seen it there....

Trying to use IMG.CFG to setup for a disk that is single density track 0 and double for the rest.

Specifically : 
Track 0 is single density (fm), with 10 x 256 byte sectors, starting at id 0
Tracks 1..79 are double density (mfm), with 16 x 256 byte sectors, starting at id 0

img.cfg contains :

[beta::326144]
cyls=80
heads=1
step=1
file-layout=interleaved

tracks=0
track-list=0
head=0
secs=10
bps=256
id=0
mode=fm

tracks=1-80
head=0
secs=16
bps=256
id=0
mode=mfm

And the disk image is called os9-boot.beta.dsk
But disk seem to be being treated as double density all the way through, I know this because the target machine has different devices
dependent on the format, /d0../d3 for the above format and /f0..f3 for standard CoCo / Dragon format disks which are dd,18x256,id=1
and even with the above defined I can only access the disk as /f2 rather than /d2 (my gotek is jumpered as drive2)

Any clues as to what's wrong? Disk image is single sided and is 326144 bytes in size

As a suggestion would be good if the documentatation contained an example of a complete img.cfg file as it's not (to me) completely
clear how it should be formatted.

Cheers.

Phill.

**Comment by keirf** [comment]
There are examples under examples/Host in the release package.

I see ateast that the track-list= line is bogus and should be removed. You might want to explicitly specify rate= for the two track types.

**Reply by prime6809** [reply]
I've removed track-list and specified a rate for both, I've also chaned head= to h= which is what the documentation says it should be.

Just to confirm the above specified filename of os9-boot.beta.dsk should match the [beta::326144] ?

A listing of the files on the usb disk : 

20/03/2022  12:27         1,982,976 beta-c-compiler_scp.hfe
20/03/2022  12:28         1,982,976 beta-os9-graphics-demo_scp.hfe
20/03/2022  12:28         1,982,976 beta-os9-lv1_2_scp.hfe
20/03/2022  11:10         1,982,976 beta-os9-master_scp.hfe
20/03/2022  12:30         1,982,976 beta-pascal_scp.hfe
20/03/2022  12:31         1,982,976 betatest_scp.hfe
01/01/2019  00:00                17 IMAGE_A.CFG
09/01/2006  00:39           737,280 nos96809l1v030206_dalpha_80d.dsk
06/05/2022  22:10           737,280 beta1.dsk
04/06/2022  19:46               219 IMG.CFG
03/06/2022  20:21           326,144 os9-boot.beta.dsk

Cheers.

Phill.

**Reply by keirf** [reply]
All sounds pretty plausible
